server.go

```diff
@@ -324,8 +324,8 @@ func (s *Server) useTransportAuthenticator(rawConn net.Conn) (net.Conn, credenti
 // Serve accepts incoming connections on the listener lis, creating a new
 // ServerTransport and service goroutine for each. The service goroutines
 // read gRPC requests and then call the registered handlers to reply to them.
-// Serve returns when lis.Accept fails. lis will be closed when
-// this method returns.
+// Serve returns when lis.Accept fails due to a non temporary network error.
```

Comment:
Err, how about `Serve returns when lis.Accept fails except for temporary network errors`, or `Serve returns a non temporary network error when lis.Accept fails`? Which do you prefer, or do you suggest any one?